http_client/views/http_client_auth_views.xml

```diff
@@ -78,12 +78,16 @@
                             invisible="auth_method == 'none'"
                             readonly="state != 'draft'"
                         >
-                            <field name="identifier" readonly="state != 'draft'"/>
-                            <field
-                                name="secret"
-                                password="True"
-                                readonly="state != 'draft'"
-                            />
+                            <field name="identifier" readonly="state != 'draft'" invisible="auth_method == 'api_key'"/>
+                            <label for="secret" invisible="auth_method == 'api_key'"/>
+                            <label for="secret" invisible="auth_method != 'api_key'"  string="API Key"/>
```

Comment:
Lets not add these workarounds. secret can be used for multiple methods and technically it can be some other kind of "secret", but it is still a secret. Lets keep it simple and have same label (in search and anywhere else, this will still be secret, so it will just add more to confusion). 

Instead I think it would be more neat to have auto generated "description" (possibly HTML), where it would provide you some explanation how to set up each of the auth methods. Let say if you specify `api_key` it would say, you need to use `Identifier` as `Api Key Header` and `Secret` as `Api Key`. 

Then it would be more universal and we would not need to resort to such workarounds:)